src/check.rs

```diff
@@ -94,42 +195,183 @@ fn check_http_url(url: &Url, ctx: &CheckContext) -> Result<(), CheckError> {
     }
 
     #[test]
     fn test_file_path() {
-        test_check_file_url("tests/html/index.html");
+        test_check_file_url("tests/html/index.html").unwrap();
     }
 
     #[test]
     fn test_directory_path() {
-        test_check_file_url("tests/html/");
+        test_check_file_url("tests/html/").unwrap();
+    }
+
+    #[test]
+    fn test_anchors() {
+        test_check_file_url("tests/html/anchors.html#h1").unwrap();
+    }
+
+    #[test]
+    fn test_missing_anchors() {
+        match test_check_file_url("tests/html/anchors.html#nonexistent") {
+            Err(CheckError::Anchor(url, fragment)) => {
+                assert!(url
+                    .as_str()
+                    .ends_with("tests/html/anchors.html#nonexistent"));
+                assert_eq!("nonexistent", fragment);
+            }
+            x => panic!(
+                "Expected to report missing anchor (Err(CheckError::FileAnchor)), got {:?}",
+                x
+            ),
+        }
+    }
+
+    #[test]
+    fn test_range_anchor() {
+        test_check_file_url("tests/html/range.html#2-4").unwrap();
+    }
+
+    #[test]
+    fn test_missing_range_anchor() {
+        match test_check_file_url("tests/html/range.html#4-6") {
+            Err(CheckError::AnchorRange(url, fragment, parts)) => {
+                assert!(url.as_str().ends_with("tests/html/range.html#4-6"));
+                assert_eq!("4-6", fragment);
+                assert_eq!(parts.len(), 1);
+                assert!(parts.contains(&"6".to_string()));
+            }
+            x => panic!(
+                "Expected to report missing anchor (Err(CheckError::FileAnchorRange)), got {:?}",
+                x
+            ),
+        }
+    }
+
+    #[test]
+    fn test_is_available_file_path() {
+        is_available(
+            &url_for("tests/html/index.html#i1"),
+            &CheckContext { check_http: false },
+        )
+        .unwrap();
+    }
+
+    #[test]
+    fn test_is_available_directory_path() {
+        is_available(
+            &url_for("tests/html/#i1"),
+            &CheckContext { check_http: false },
+        )
+        .unwrap();
+    }
+
+    #[test]
+    fn test_missing_dir_index_fragment() {
+        match is_available(
+            &url_for("tests/html/missing_index/#i1"),
+            &CheckContext { check_http: false },
+        ) {
+            Err(CheckError::File(path)) => assert!(path.ends_with("tests/html/missing_index")),
+            x => panic!(
+                "Expected to report missing anchor (Err(CheckError::File)), got {:?}",
+                x
+            ),
+        }
+    }
+
+    #[test]
+    fn test_http_check() {
+        let root = mock("HEAD", "/").with_status(200).create();
+
+        let mut url = mockito::server_url();
+        url.push_str("/");
+
+        is_available(
+            &Url::parse(&url).unwrap(),
+            &CheckContext { check_http: true },
+        )
+        .unwrap();
+
+        root.assert();
```

Comment:
Hmm, I'm not a giant fan of having to call this explicitly - do you know if it runs on drop? If not, oh well, but it would be nice to have it done automatically.

Reply:
Looking at the docs, and briefly at the source code, doesn't seem like it is. I've tried adding a mocked response that will not be requested and test did not fail. So this is required I'm afraid.

Reply:
Fair enough, it's not the end of the world.